### PR TITLE
cleanup `index.js` and drop `0.12` node

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -111,7 +111,7 @@
     "ember-inflector": "^2.0.0"
   },
   "engines": {
-    "node": ">= 0.12.0"
```
Comment (rwjblue) [inline package.json]: FWIW, in practice we have already done this bump (we include deps that are only Node 4+ compatible since at least 2.12).

Comment (bekzod) [inline package.json]: thanks for clarifying, sorry I wrote bad commit message, we still should keep `engines` section right ?

Comment (rwjblue) [inline package.json]: Yep, the changes are good. I was just commenting to make it clear to the other maintainers that Node 4 was essentially already required...

**Comment by bmac** [comment]
Thanks @bekzod.

**Pull request merged**